### PR TITLE
board generator documentation

### PR DESCRIPTION
**Pull request by d-a-v**
Changes can be seen [there](https://github.com/d-a-v/Arduino/blob/boardsdoc/doc/faq/readme.rst).

### CODE REVIEW

File: doc/faq/a02-my-esp-crashes.rst [hunk 1]
```diff
@@ -286,6 +286,9 @@ Memory, memory, memory
    modification. For example, std::vector is available for use. The standard implementations
    rely on exceptions for error handling, which is not available for the ESP, and in any
    case there is no access to the underlying code.
+   
+   Instrumenting the code with the OOM debug option and calls to ``ESP.getFreeHeap()`` will
```
Comment (earlephilhower) [inline doc/faq/a02-my-esp-crashes.rst]: Should we add a line about the last memory allocation failure report in the panic that was added a while back?

File: doc/faq/a05-board-generator.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
+:orphan:
+
+Board generator
+---------------
+
+The board generator is a python script originally intended to ease the
+Arduino IDE's `boards.txt` configuration file about the multitude of
+available board, especially when common data have to be updated for all of
```
Comment (devyte) [inline doc/faq/a05-board-generator.rst]: available boards

Comment (devyte) [reply doc/faq/a05-board-generator.rst]: common data -> common parameters

File: doc/faq/a05-board-generator.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+-  `How can I run the script ? <#how-can-i-run-the-script>`__
+-  `What can I do with it ? <#what-can-i-do-with-it>`__
+-  `When do I need to update it ? <#when-do-i-need-to-mess-with-it>`__
+-  `Why is my pull-request failing continuous-integration ? <#why-is-my-pull-request-failing-continuous-integration>`__
+
+How can I run the script ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Python 2 needs to be installed on your system.
```
Comment (devyte) [inline doc/faq/a05-board-generator.rst]: Does it work with Python 3?

File: doc/faq/a05-board-generator.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+As of today you can:
+
+* in the IDE: change the default serial programming speed of any board
+
+* in the IDE: add new serial programming speed
+
+* increase available flash space by disabling floats in ``*printf`` functions
+
+* enable WPS which is now disabled by default (at the cost of a smaller heap by ~4KB)
```
Comment (devyte) [inline doc/faq/a05-board-generator.rst]: in the IDE?

Comment (devyte) [reply doc/faq/a05-board-generator.rst]: all of these are in the IDE, right?

File: doc/faq/a05-board-generator.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+* change led pin ``LED_BUILTIN`` for the two generic boards
+
+* change the default lwIP version (1.4 or 2)
+
+
+When do I need to mess with it ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The board generator is used to automate configuration files generation when it is
```
Comment (devyte) [inline doc/faq/a05-board-generator.rst]: automate generation of configuration files

File: doc/faq/a05-board-generator.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+possible to. It needs to be edited for:
+
+* All the specific board informations.  This is the only place where a new
```
Comment (devyte) [inline doc/faq/a05-board-generator.rst]: All information for specific boards

File: doc/faq/readme.rst [hunk 7]
```diff
@@ -139,3 +145,17 @@ The following lines are compatible with both lwIP versions:
...
 
 Ref.  `#1923 <https://github.com/esp8266/Arduino/issues/1923>`__
+
+
+Why is there a board generator and what about it ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The board generator is a python script originally intended to ease the
+Arduino IDE's `boards.txt` configuration file about the multitude of
+available board, especially when common data have to be updated for all of
```
Comment (devyte) [inline doc/faq/readme.rst]: available boards

Comment (devyte) [reply doc/faq/readme.rst]: common data -> common parameters

File: doc/faq/readme.rst [hunk 8]
```diff
@@ -139,3 +145,17 @@ The following lines are compatible with both lwIP versions:
...
+them.
+
+This script is also used to manage uncommon options that will not find their
```
Comment (devyte) [inline doc/faq/readme.rst]: that are currently not available in the IDE menu (we can't say for sure that they will never ever be in the IDE :p)

File: doc/faq/a05-board-generator.rst (hunk 3 above)
Comment (d-a-v) [reply doc/faq/a05-board-generator.rst]: > Does it work with Python 3?

No. At least all `print`s must be updated.

File: doc/faq/a05-board-generator.rst (hunk 4 above)
Comment (d-a-v) [reply doc/faq/a05-board-generator.rst]: float/printf and WPS are not in IDE. The board generator script is currently the only way to change these.

**Pull request merged**